executor/src/lib.rs

```diff
@@ -62,12 +64,12 @@ pub trait ExecutionState {
     /// Execute the transaction and atomically persist the consensus index. This function
     /// returns an execution outcome that will be output by the executor channel. It may
     /// also return a new committee to reconfigure the system.
-    async fn handle_consensus_transaction<PublicKey: VerifyingKey>(
+    async fn handle_consensus_transaction(
         &self,
-        consensus_output: &ConsensusOutput<PublicKey>,
+        consensus_output: &ConsensusOutput<Self::PubKey>,
```

Comment:
this `ConsensusOutput` struct means something different in sui vs narwhal:

```
pub struct ConsensusOutput {
    #[serde(with = "serde_bytes")]
    pub message: Vec<u8>,
    pub sequence_number: SequenceNumber,
}
```

in narwhal
```
pub struct ConsensusOutput<PublicKey: VerifyingKey> {
    /// The sequenced certificate.
    pub certificate: Certificate<PublicKey>,
    /// The (global) index associated with this certificate.
    pub consensus_index: SequenceNumber,
}
```

Reply:
We will indeed have no other choice, but that argument can simply be ignored until it is useful.